From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: deepak1556 <hop2deep@gmail.com>
Date: Wed, 19 Dec 2018 08:54:45 +0530
Subject: fix: use bssl BN_bn2bin_padded for ossl's BN_bn2binpad

Also note that BN_bn2bin_padded returns 1 on success instead of size

diff --git a/src/node_crypto.cc b/src/node_crypto.cc
index 590c6d1c374c086e128580d6160ccbd71a7ef227..4933bf7f1bd2cd5e4293e52a241fcfd5fda6b779 100644
--- a/src/node_crypto.cc
+++ b/src/node_crypto.cc
@@ -5408,9 +5408,9 @@ void DiffieHellman::GenerateKeys(const FunctionCallbackInfo<Value>& args) {
   const int size = BN_num_bytes(pub_key);
   CHECK_GE(size, 0);
   AllocatedBuffer data = env->AllocateManaged(size);
-  CHECK_EQ(size,
-           BN_bn2binpad(
-               pub_key, reinterpret_cast<unsigned char*>(data.data()), size));
+  CHECK_EQ(
+      1,
+      BN_bn2bin_padded(reinterpret_cast<unsigned char*>(data.data()), size, pub_key));
   args.GetReturnValue().Set(data.ToBuffer().ToLocalChecked());
 }
 
@@ -5430,8 +5430,8 @@ void DiffieHellman::GetField(const FunctionCallbackInfo<Value>& args,
   CHECK_GE(size, 0);
   AllocatedBuffer data = env->AllocateManaged(size);
   CHECK_EQ(
-      size,
-      BN_bn2binpad(num, reinterpret_cast<unsigned char*>(data.data()), size));
+      1,
+      BN_bn2bin_padded(reinterpret_cast<unsigned char*>(data.data()), size, num));
   args.GetReturnValue().Set(data.ToBuffer().ToLocalChecked());
 }
 
@@ -5748,9 +5748,9 @@ void ECDH::GetPrivateKey(const FunctionCallbackInfo<Value>& args) {
 
   const int size = BN_num_bytes(b);
   AllocatedBuffer out = env->AllocateManaged(size);
-  CHECK_EQ(size, BN_bn2binpad(b,
-                              reinterpret_cast<unsigned char*>(out.data()),
-                              size));
+  CHECK_EQ(1, BN_bn2bin_padded(reinterpret_cast<unsigned char*>(out.data()),
+                                  size,
+                                  b));
 
   Local<Object> buf = out.ToBuffer().ToLocalChecked();
   args.GetReturnValue().Set(buf);
